core/controllers/blog_admin.py

```diff
@@ -106,33 +116,52 @@ def get(self) -> None:
     @acl_decorators.can_access_blog_admin_page
     def post(self) -> None:
         """Handles POST requests."""
-        result = {}
-        if self.normalized_payload.get(
-                'action') == 'save_config_properties':
-            new_config_property_values = self.normalized_payload.get(
-                'new_config_property_values')
+        assert self.user_id is not None
+        payload_data = cast(
+            BlogAdminHandlerNormalizedPayloadDict, self.normalized_payload
+        )
+        if payload_data['action'] == 'save_config_properties':
+            new_config_property_values = payload_data[
+                'new_config_property_values']
+            if new_config_property_values is None:
+                raise Exception(
+                    'The new_config_property_values cannot be None when the'
+                    ' action is save_config_properties.'
+                )
```

Comment:
Added a todo https://github.com/oppia/oppia/issues/16364. In the code also!

Reply:
@vojtechjelinek @sahiljoster32 I think having a single handler should be fine as both the actions, 'reverting' and 'saving' change the value of the config property i.e  update the value of config property in the backend. Having separate handler would be more reasonable if the type of action being performed would have been different. We follow the same logic in admin.py for saving and reverting config property:  https://github.com/oppia/oppia/pull/16342/files#diff-981a56d20653e10899f2301ebcb06864bda37939c907e36965e2840ae1c36cbbR323  and https://github.com/oppia/oppia/pull/16342/files#diff-981a56d20653e10899f2301ebcb06864bda37939c907e36965e2840ae1c36cbbR336.

Reply:
Looking at the reasoning and example I think it is ok to keep the current handler but It would be great if @seanlip @vojtechjelinek confirm it as well. Thanks!

Reply:
I partially agree. The thing is that we should differentiate between the different actions on the same object by using different HTTP handlers (like PUT and DELETE). This is better because it enables us to do more precise handler schema validation and also this will help with more proper typing (in the future).

Reply:
But in this case, both operations are logically PUTs, right? (according to REST) since they modify the data. I don't think reverting is the same as a delete operation.

Reply:
Yeah, I think you're right. But the issue still stays I think. We cannot really easily validate schema in the handlers that use `action` (and also MyPy). For some the solution might be to split them up (for example in the preferences page or in _admin.py_ (where `action` is used just to reduce the number of handlers).